src/com/perl5/lang/perl/idea/run/PerlRunProfileState.java

```diff
@@ -155,4 +163,54 @@ protected ProcessHandler startProcess() throws ExecutionException
 		return runProfile.getEnvs();
 	}
 
+	public List<String> getPerlLibraries(@Nullable VirtualFile scriptFile)
+	{
+		if (!PlatformUtils.isIntelliJ())
+		{
+			return Collections.emptyList();
+		}
+		else
+		{
+			PerlConfiguration runProfile = (PerlConfiguration) getEnvironment().getRunProfile();
+
+			// try to get used SDK
+			Sdk sdk;
+			if (runProfile.isUseAlternativeSdk())
+			{
+				sdk = ProjectJdkTable.getInstance().findJdk(runProfile.getAlternativeSdkPath());
+			}
+			else
+			{
+				Module moduleForFile = scriptFile == null ? null : ModuleUtilCore.findModuleForFile(scriptFile, getEnvironment().getProject());
+
+				sdk = PerlRunUtil.getModuleSdk(moduleForFile);
+				if (sdk == null)
+				{
+					// try project SDK
+					sdk = ProjectRootManager.getInstance(getEnvironment().getProject()).getProjectSdk();
+				}
+			}
+
+			if (sdk != null && sdk.getSdkType() == PerlSdkType.getInstance())
+			{
+				List<String> incPaths = PerlSdkType.getInstance().getINCPaths(sdk.getHomePath());
+				List<String> list = new ArrayList<>();
+
+				for (VirtualFile file : sdk.getRootProvider().getFiles(OrderRootType.CLASSES))
```

Comment:
Have you tried this ? It looks to me that you are taking classroots from sdk, not from module or project. I'm still thinking that `LibraryUtil` is best choice here. It will take all classroot from module except `@INC` ones (which are JDK classroots) with false last argument. Probably this will work if you are using current project JDK but sure won't if use alternative one. 

But, you still need to exclude `com.perl5.lang.perl.idea.configuration.settings.PerlSharedSettings#getLibRootUrls` ones, which added in `com.perl5.lang.perl.util.PerlRunUtil#getPerlCommandLine`

Dam, it so painful to see my own old code :)